grc/gui/MainWindow.py

```diff
@@ -246,12 +246,29 @@ def new_page(self, file_path='', show=False):
             if file_path: Messages.send_start_load(file_path)
             flow_graph = self._platform.make_flow_graph()
             flow_graph.grc_file_path = file_path
-            #print flow_graph
+
+            if not file_path or self._platform.is_trusted(file_path):
+                flow_graph.view_only = False
```

Comment:
Shouldn't this be already handled in `make_flow_graph()`?

Reply:
MainWindow doesn't pass a file path to `make_flow_graph` which means `is_trusted` will fail in `make_flow_graph` but succeed in MainWindow (if trusted). So `make_flow_graph` has the check for the times a path is provided directly, but in the case of MainWindow it is checked outside (as you wouldn't want to pass the path since it would parse+import as well).

```python
def make_flow_graph(self, from_filename=None):
    fg = self.FlowGraph(parent=self)
    if not self.view_only_mode or self.is_trusted(from_filename):
        fg.view_only = False
    if from_filename:
        data = self.parse_flow_graph(from_filename)
        fg.grc_file_path = from_filename
        fg.import_data(data)
    return fg
```
Though, there may be a more elegant solution. I'll try to think of one!